### PR TITLE
added new animal form and tests

### PR DESCRIPTION
**Pull request by tnguyen3645**
Changes:

- Added view for new animal form
- Added AnimalController with new and create methods
- Added spec tests for saving new animal
- Refactored routes.rb

### CODE REVIEW

**Comment by tnguyen3645** [comment]
When I run `bundle install` I get a different platform in my Gemfile.lock compared to that of on main. Not sure how to resolve this merge conflict.

File: app/controllers/animals_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    if @animal.save
+      flash[:success] = "Animal added successfully"
+      redirect_to animal_path(@animal.id)
+    else
+      flash.now[:error] = @animal.errors.full_messages.to_sentence
+      render :new
+    end
+  end
+
+  private
```
Comment (nwalberts) [inline app/controllers/animals_controller.rb:26]: lets add a space after the private keyword here

File: app/controllers/animals_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class AnimalsController < ApplicationController
+  def index
+    render 'static_pages/index'
+  end
+
+  def show
+    render 'static_pages/index'
+  end
+
```
Comment (nwalberts) [inline app/controllers/animals_controller.rb:9]: this is totally fine, if you wanted to you could also in routes.rb have requests with these paths go immediately to the `static_pages_controller#index`. You will likely be doing that often anyways. Up to you, just try to be consistent

File: spec/features/show_page/show_page_spec.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+feature 'user adds a new animal' do
+  scenario 'user does not fill form and tries to submit' do
+    visit new_animal_path
+
+    click_button 'Add Animal'
+    expect(page).to have_content("Name can't be blank")
+    expect(page).to have_content("Body can't be blank")
+  end
+end
```
Comment (nwalberts) [inline spec/features/show_page/show_page_spec.rb:11]: Im a bit confused with this spec
- the name of the file says show, but this isnt for a show page
- you only are testing the sad path here, not the happy path as well 

Try to make this test more robust, and update the filename

File: spec/features/user_creates_new_animal_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+          name: "dog",
+          body: "dope"
+        }
+      }
+
+      prev_count = Animal.count
+      post(:create, params: post_json, format: :json)
+      expect(Animal.count).to eq(prev_count + 1)
+      expect(Animal.where(name: "dog")).to exist
+    end
```
Comment (nwalberts) [inline spec/features/user_creates_new_animal_spec.rb:17]: this happens to be passing, but is not necessary, and doesnt actually test the functionality the way it is intended

- we dont actually send json to this endpoint. This endpoint is a non-api controller, that receives standard params.
- We dont need a controller test for it, the capybara test is fine for features that are not made with react
We can likely delete this

**Review by nwalberts — COMMENTED** [review]
this would be the changes requested pre-merge. Please prioritize remedying these comments before proceeding

**Pull request merged**